lib/os.sh

```diff
@@ -10,25 +10,43 @@ function which-os {
   fi
 }
 
-function guess-ipaddr {
-  if [ $(which-os) == "darwin" ]; then
-    ifconfig vboxnet2 | grep 'inet ' | awk '{print $2}'
-  else
-    ifconfig en0 | grep 'inet ' | awk '{print $2}'
+function guess-docker-ipaddr {
+  if [ -n "${HOST_IPADDR:-}" ]; then
+    echo ${HOST_IPADDR}
+    return 0
   fi
-}
 
-function guess-registry {
-  local ip
+  case $(which-os) in
+    darwin)
+      boot2docker config | grep "HostIP" | cut -d = -f 2 | xargs
```

Comment:
We recommend that users go with `docker-machine` now, not `boot2docker` (although yours truly finally switched over only yesterday). So that should now be the correct way to query such things. Maybe we could try `docker-machine`, then `boot2docker`? Here's some shell that gets us close with `docker-machine`:
```console
$ docker-machine env `docker-machine active` | grep DOCKER_HOST | cut -d = -f 2
"tcp://192.168.99.101:2376"
```